core/src/main/java/hudson/tasks/Maven.java

```diff
@@ -137,6 +137,12 @@
      */
     private GlobalSettingsProvider globalSettings;
 
+    /**
+     * Inject build variables as properties into maven process.
+     * @since TODO
+     */
+    private boolean injectBuildVariables = true;
```

Comment:
taking a *quick* peak at the code it looks like we should already set environment variables (unless I missed something)